### DISCUSSION TITLE
Successful Make but not executable cardinal-opt

### DISCUSSION THREAD

**Discussion by AnthonyB08** [neams-th-coe/cardinal#609] [Q&A]
Hello all, 

I am trying to install cardinal on my friend's computer so we can work on a project together. I went through all the same steps as I did for my machine, and was able to complete a make -j8 in cardinal's top directory. Everything compiles successfully, but when we go to run tests he has no cardinal-opt executable file in the cardinal directory. 

What does the community suggest we do to correct this?

**Comment by aprilnovak** [comment]
If there's no `cardinal-opt`, then the compilation wasn't successful - sometimes the error message will get buried by other screen output. Can you please re-run make, but send the output to a text file and post it here so I can take a look?

```
make -j8 > output.txt
```

**Comment by AnthonyB08** [comment]
[output.txt](https://github.com/neams-th-coe/cardinal/files/10828481/output.txt)

**Comment by aprilnovak** [comment]
I see that something went wrong with MOAB:

```
-- Configuring incomplete, errors occurred!
See also "/home/jamesf03/cardinal/build/moab/CMakeFiles/CMakeOutput.log".
See also "/home/jamesf03/cardinal/build/moab/CMakeFiles/CMakeError.log".
```

Can you please attach those two CMake output files here as well?

**Reply by AnthonyB08** [reply]
Yes here you 

[CMakeOutput.log](https://github.com/neams-th-coe/cardinal/files/10828637/CMakeOutput.log)
[CMakeError.log](https://github.com/neams-th-coe/cardinal/files/10828638/CMakeError.log)

**Reply by aprilnovak** [reply]
In CMakeError, I'm seeing errors related to `std::isfinite` not existing... Can you tell me a bit more about this system? Can you show me

```
env | sort
```

I'd also like to try and see if disabling MOAB will fix things (which can then give us a clue). Can you do

```
cd cardinal
rm -rf build/ install
export ENABLE_DAGMC=false
make -j8
```

**Comment by AnthonyB08** [comment]
[output3.txt](https://github.com/neams-th-coe/cardinal/files/10830059/output3.txt)
 I will post his compiling results, with dagmc = false, tomorrow

**Reply by AnthonyB08** [reply]
He was able to compile successfully which is strange because I was able to compile with DAGMC = true. I will work on neutronics later this evening to fully ensure it is enabled.

**Reply by aprilnovak** [reply]
Is this build attempting with ENABLE_DAGMC, ENABLE_NEK, and ENABLE_OPENMC all set to true? Because if that is the case, then the MOOSE conda environment does not support the Nek part.

If you have Nek disabled, then you can use MOOSE's conda environment. I just checked on my Mac, and was able to build with ENABLE_DAGMC and ENABLE_OPENMC with MOOSE's conda.

**Comment by AnthonyB08** [comment]
His build was without conda so NEKS, OpenMC , and DAGMC were enabled. He was only to compile by disabling DAGMC. I installed cardinal to my home PC, using the Conda method, and was not able to compile with DAGMC Enabled.

**Reply by aprilnovak** [reply]
We very recently added DAGMC to our build, so it's likely that you are the first users to try it out. It's very possible that our build settings are missing something. @pshriwise is our DAGMC/Moab expert - Patrick, does anything jump out to you with the errors being seen here? To me, it looks like the error is `std::isfinite` not existing (but needed by Moab). Is there a minimum compiler requirement in Moab we should be enforcing?

**Reply by AnthonyB08** [reply]
That makes sense, I will run some DAGMC code this evening on my laptop since it WAS able to compile with DAGMC enabled and give some updates

**Reply by aprilnovak** [reply]
We hope to have a conda package for Cardinal soon, so we'll have a more elegant fix in case there's some persistent issue with building with DAGMC that is platform-dependent. 

There's not any significant difference in Cardinal usage with/without DAGMC, so for getting-started purposes it should be fine to proceed without DAGMC (we should have a solution soon).

**Comment by pshriwise** [comment]
It looks to me like the CMake macro `CHECK_FUNCTION_EXISTS` is creating an invalid source file and trying to compile it to test for these functions. I'm not sure why though.

I'm saying this because of this line:

```
/usr/bin/mpicc   -fopenmp -fPIC -DPIC -DCHECK_FUNCTION_EXISTS=std::isfinite -o CMakeFiles/cmTC_88b97.dir/CheckFunctionExists.c.o -c /home/jamesf03/cardinal/build/moab/CMakeFiles/CMakeScratch/TryCompile-Lm4bI4/CheckFunctionExists.c
<command-line>: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘::’ token
```

The `<command-line>` error appears to be indicating a syntax error in `CheckFunctionExists.c`. So `std::isfinite` may be present (I'd expect that it is), but CMake isn't able to compile this test program to find out. I'm not sure why though. 

Could you paste the contents of `/home/jamesf03/cardinal/build/moab/CMakeFiles/CMakeScratch/TryCompile-Lm4bI4/CheckFunctionExists.c` here if possible?

**Reply by AnthonyB08** [reply]
Hi pshriwise! (You are on all the forums haha)

James does not have a moab folder because DAGMC had to be disabled for a successful compile.
James build/ directory:
![image](https://user-images.githubusercontent.com/55299169/221907500-b47ddb5e-4702-4952-8de3-0f499a22e666.png)

 My laptop was able to compile successfully with DAGMC enabled, strangely enough, but there is no TryComple-Lm4bI4 folder. These are the contents in the CMakeScratch directory.

**Reply by AnthonyB08** [reply]
![image](https://user-images.githubusercontent.com/55299169/221907950-a14a232a-40d8-4fa3-be49-d1d04c41830c.png)

**Reply by aprilnovak** [reply]
Right, I think Patrick is asking to go through the steps to try building with ENABLE_DAGMC=true, and then paste the contents of `/home/jamesf03/cardinal/build/moab/CMakeFiles/CMakeScratch/TryCompile-Lm4bI4/CheckFunctionExists.c`.

**Reply by AnthonyB08** [reply]
Ah my I miss understood, I will do that this evening.

**Reply by AnthonyB08** [reply]
Sorry for the delay. Going through the steps with ENABLE_DAGMC=true there are no contents in the respective directory.

**Comment by AnthonyB08** [comment]
Update: I do not know how but I managed to compile with DAGMC enabled on my laptop. Later this week I will follow the same steps and update this discussion. 
![image](https://user-images.githubusercontent.com/55299169/223516854-89b4e76c-2081-4be5-9335-9b4657da68c0.png)

I am very excited to have multiphysics under my belt , soon!
